pkg/process/metadata/parser/service.go

```diff
@@ -227,6 +243,36 @@ func isRuneLetterAt(s string, position int) bool {
 	return len(s) > position && unicode.IsLetter(rune(s[position]))
 }
 
+// extractEnvsFromCommand separates the env var declaration from the command + args part
+func extractEnvsFromCommand(cmd []string) ([]string, []string) {
+	pos := 0
+	for _, arg := range cmd {
+		if !strings.ContainsRune(arg, '=') {
+			break
+		}
+		pos++
+	}
+	return cmd[:pos], cmd[pos:]
+}
+
+// chooseServiceNameFromEnvs extracts the service name from usual tracer env variables (DD_SERVICE, DD_TAGS).
+// returns the service name, true if found, otherwise "", false
+func chooseServiceNameFromEnvs(envs []string) (string, bool) {
+	for _, env := range envs {
+		if strings.HasPrefix(env, "DD_SERVICE=") {
+			return strings.TrimPrefix(env, "DD_SERVICE="), true
+		} else if strings.HasPrefix(env, "DD_TAGS=") && strings.Contains(env, "service:") {
```

Comment:
nit: you don't need the `else` (just a `if`) since you are returning in the previous `if` clause.

Reply:
Do you have to go over all the tags? 
```go
i := strings.Index(env, "service:")
if i >= 0 {
  j := strings.IndexRune(env[i:], ",")
  if j < 0 {
    j = len(env)
  }
  return strings.TrimPrefix(env[i:j], "service:"), true
}
```

Reply:
That's for sure more efficient. I can use it thanks

Reply:
edit: I prefer to use the original implementation even if inefficient. For instance that approach can give false positive if the tag has a suffix `service` (not a whole word). i.e. `peer_service:mysvc` (it's a real world tag)